include/boost/math/tools/roots.hpp

```diff
@@ -213,6 +213,245 @@ inline std::pair<T, T> bisect(F f, T min, T max, Tol tol) noexcept(policies::is_
+      static_assert(!std::is_same<T, float>::value, "Need to use Midpoint754 solver when T is float");
+      static_assert(!std::is_same<T, double>::value, "Need to use Midpoint754 solver when T is double");
+
+   public:
+      static T solve(T x, T X) {
+         const T sx = sign(x);
+         const T sX = sign(X);
+         
+         // Sign flip return zero
+         if (sx * sX == -1) { return T(0.0); }
+
+         // At least one is positive
+         if (0 < sx + sX) { return do_solve(x, X); }
+
+         // At least one is negative
+         return -do_solve(-x, -X);
+      }
+
+   private:      
+      struct EqZero {
+         EqZero(T x) { BOOST_MATH_ASSERT(x == 0 && "x must be zero."); }
+      };
+
+      struct EqInf {
+         EqInf(T x) { BOOST_MATH_ASSERT(x == static_cast<T>(std::numeric_limits<double>::infinity()) && "x must be infinity."); }
+      };
+
+      class PosFinite {
+      public:
+         PosFinite(T x) : x_(x) {
+            BOOST_MATH_ASSERT(0 < x && "x must be positive.");
+            BOOST_MATH_ASSERT(x < std::numeric_limits<float>::infinity() && "x must be less than infinity.");
+         }
+
+         T value() const { return x_; }
+
+      private:
+         T x_;
+      };
+
+      // Two unknowns
+      static T do_solve(T x, T X) {
+         if (X < x) {
+            return do_solve(X, x);
+         }
+
+         if (x == 0) {
+            return do_solve(EqZero(x), X);
+         } else if (x == static_cast<T>(std::numeric_limits<double>::infinity())) {
+            return static_cast<T>(std::numeric_limits<double>::infinity());
+         } else {
+            return do_solve(PosFinite(x), X);
+         }
+      }
+
+      // One unknowns
+      static T do_solve(EqZero x, T X) {
+         if (X == 0) {
+            return T(0.0);
+         } else if (X == static_cast<T>(std::numeric_limits<double>::infinity())) {
+            return T(1.0);
+         } else {
+            return do_solve(x, PosFinite(X));
+         }
+      }
+      static T do_solve(PosFinite x, T X) {
+         if (X == static_cast<T>(std::numeric_limits<double>::infinity())) {
+            return do_solve(x, EqInf(X));
+         } else {
+            return do_solve(x, PosFinite(X));
+         }
+      }
+
+      // Zero unknowns
+      template <typename U = T>
+      static typename std::enable_if<std::numeric_limits<U>::is_specialized, T>::type
+      do_solve(PosFinite x, EqInf X) {
+          return do_solve(x, PosFinite((std::numeric_limits<U>::max)()));
+      }
+      template <typename U = T>
+      static typename std::enable_if<!std::numeric_limits<U>::is_specialized, T>::type
+      do_solve(PosFinite x, EqInf X) {
+          BOOST_MATH_ASSERT(false && "infinite bounds support requires specialization.");
+          return static_cast<T>(std::numeric_limits<T>::signaling_NaN());
+      }
+
+      template <typename U = T>
+      static typename std::enable_if<std::numeric_limits<U>::is_specialized, U>::type
```

Comment:
We want to be able to bisect with denormals because the solution to the root finding problem could be a denormal number.

Reading the [depreciation document](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2614r2.pdf), it seems like the depreciation of `has_denorm` has a lot to do with it being a `static constexpr`. This is problematic because the actual behavior can change at runtime with e.g., `_MM_SET_FLUSH_ZERO_MODE`.

I think we can actually delete `std::numeric_limits::min()` because `denorm_min` returns `numeric_limits::min()` if denormals are off. Does that seen sensible?

Reply:
> 
> I think we can actually delete `std::numeric_limits::min()` because `denorm_min` returns `numeric_limits::min()` if denormals are off. Does that seen sensible?

I think so. Since you are already check for `is_specialized` you should avoid ever getting 0 instead of a finite minimum value.

Reply:
> I think so. Since you are already check for `is_specialized` you should avoid ever getting 0 instead of a finite minimum value.

https://github.com/boostorg/multiprecision/pull/562